shared/src/federatedIdentities/federatedIdentity.dto.ts

```diff
@@ -10,26 +10,36 @@ export const federatedIdentityProviders = [
 ] as const;
 
 type GenericFederatedIdentity<
-  P extends FederatedIdentityProvider,
+  Provider extends FederatedIdentityProvider,
   T extends PeConnectToken | InclusionConnectJwt,
+  P,
 > = {
-  provider: P;
+  provider: Provider;
   token: T;
+  payload?: P;
 };
 
 export const authFailed = "AuthFailed";
 export const notJobSeeker = "NotJobSeeker";
 
 export type PeExternalId = Flavor<string, "PeExternalId">;
 
+type PeConnectAdvisorDtoInFront = {
```

Comment:
```suggestion
type PeConnectAdvisorForBeneficiary = {
```
? Je suis pas sûr de ma suggestion, mais ça ne devrait pas s'appeler "InFront" je pense (il sera utilisé ailleurs probablement).